fw/vhost.c

```diff
@@ -904,6 +926,84 @@ tfw_mod_hdr_sort(TfwLocation *loc)
 	}
 }
 
+static void
+__set_bit_range(int start, int end, unsigned long *codes)
+{
+	int i;
+
+	for (i = HTTP_CODE_BIT_NUM(start); i < HTTP_CODE_BIT_NUM(end); i++) {
+		__set_bit(i, codes);
+	}
+}
+
+static int
+tfw_cfgop_cache_use_stale(TfwCfgSpec *cs, TfwCfgEntry *ce, TfwLocation *loc)
+{
+	TfwCacheUseStale *cfg;
+	size_t i;
+	int n = 0;
+	bool mask5x = false, mask4x = false;
+
+	if (ce->attr_n) {
+		T_ERR_NL("%s arguments may not have the '=' sign\n",
+			 cs->name);
+		return -EINVAL;
+	}
+
+	if (ce->val_n < 1) {
+		T_ERR_NL("%s too few arguments\n", cs->name);
+		return -EINVAL;
+	}
+
+	cfg = kzalloc(sizeof(TfwCacheUseStale), GFP_KERNEL);
+	if (!cfg)
+		return -ENOMEM;
+	loc->cache_use_stale = cfg;
```

Comment:
This overriding only for default vhost, I don't see a reason to fix it, because default vhost will be removed soon. Now I just I forbade this directive for default vhost. 

About inheritance: it's has same logic with `req_hdr_set` and we also don't have inheritance for these directives. I think we should fix it together in another task, we have such task for inheritance.

Reply:
No error is still present.
vhost xxx {
	proxy_pass default;
	tls_certificate /home/evgeny/workdir/certs/tempesta-tech.test.crt;
	tls_certificate_key /home/evgeny/workdir/certs/tempesta-tech.test.key;
	cache_use_stale 400;
	cache_use_stale 500;
}

And we have a memory leak. Because we don't check dublicate values.

Reply:
For `http_resp_code_block`  I catch duplicate entry: 'http_resp_code_block', for such config. Moreover
```
if (cfg->http_resp_code_block) {
		if (ce->dflt_value)
			return 0;
		kfree(cfg->http_resp_code_block);
		cfg->http_resp_code_block = NULL;
	}
```

Reply:
We need also check how we handle this directive when it is present in vhost and location inside vhost

Reply:
It seems we should set .allow_repeat = false, to this option

Reply:
Fixed

Reply:
if allow_repeat is true there is a memory leak here.